nodes/config/ui_base.html

```diff
@@ -322,13 +358,19 @@
             refreshSidebarEditorDebounced()
         }
     })
-    RED.events.on('nodes:add', function (event) {
-        if (RED._db2debug) { console.log('nodes:add', event) }
-        if (event.dirty && event.type && event.type.startsWith('ui-')) {
+    RED.events.on('nodes:add', function (node) {
+        if (RED._db2debug) { console.log('nodes:add', node) }
+        if (node.dirty && node.type && node.type.startsWith('ui-')) {
             if (RED._db2debug) { console.log('nodes:add - this is a ui- node! queuing a call to refreshSidebarEditors') }
             // debounce the call to refreshSidebarEditors as multiple events can be fired in quick succession
             refreshSidebarEditorDebounced()
         }
+
+        // if we're adding a ui-base
+        if (node.type.startsWith('ui-')) {
+            // action on all ui- elements to ensure we've remapped (now) missing ui-base nodes
+            checkDuplicateUiBases()
+        }
```

Comment:
Calling the `checkDuplicateUiBases` on each 'nodes:add' _might_ be a bottleneck on a large flow.  

the 'nodes:add' function is called for every node imported.  e.g. this is from importing a flow with 7 UI nodes:
![image](https://github.com/FlowFuse/node-red-dashboard/assets/44235289/e622e53a-6019-4825-ae8f-d87ad653c859)

If a user has 5000 regular nodes in their flows (not unheard of) then we will be looping `eachNode` 7 times (7 x 5000 iterations)

We might want to consider either debouncing this -or- simply checking `node.type === 'ui-base'` and applying the current 'ui-base' (if there is one)?

Reply:
debouncing seems more sensible. The approach of checking `ui-base` type is unfortunately flawed because if we have our `ui-base` import _first_, then when we haven't got a quick win for being able to update all of the `ui` references in the other nodes?